tests/test_hook_man_generator/ExpectedHookCaller.hpp

```diff
@@ -0,0 +1,20 @@
+#include <functional>
+
+namespace HookMan {
+
+template <typename F_TYPE> std::function<F_TYPE> register_c_func(uintptr_t p) {
+    return std::function<F_TYPE>(reinterpret_cast<F_TYPE *>(p));
+}
+
+class HookCaller {
+public:
+    inline int friction_factor(int v1, int v2) { return this->_friction_factor(v1, v2); }

```

Comment:
How should we handle default functions for hooks?

Reply:
As we discussed:

```c++
std::function<int(int, int)> friction_factor() {
    reutnr this->_friction_factor;
}
```

Reply:
- [x] Applied